index.js

```diff
@@ -9,9 +8,7 @@ var http = exports
 http.request = function (opts, cb) {
 	if (typeof opts === 'string')
 		opts = url.parse(opts)
-	else
-		opts = extend(opts)
```

Comment:
This is not a no-op. It actually clones the object.

Reply:
You could do a shallow clone with Object.assign({}, opts) or spread {..opts} Where Object.assign is supported from node v4

Reply:
sorry ignore above, ie11 do not support assign